WcaOnRails/app/models/competition.rb

```diff
@@ -2405,6 +2405,28 @@ def set_form_data_series(form_data_series, current_user)
     self.competition_series = competition_series
   end
 
+  def payments_enabled?
+    competition_payment_integrations.exists?
+  end
+
+  def payment_account_for(integration_name)
+    validate_integration_name!(integration_name)
+
+    payment_integrations = competition_payment_integrations.where(
+      connected_account_type: CompetitionPaymentIntegration::AVAILABLE_INTEGRATIONS[integration_name],
+    )
+
+    return nil unless payment_integrations.count > 0
+
+    # Take the first result as `where` always returns an array
+    payment_integrations.first.connected_account
+  end
+
+  private def validate_integration_name!(integration_name)
```

Comment:
I would leave this specific method in the CPI model class and call it / reference it here.
```ruby
def payment_account_for(type)
  CompetitionPaymentIntegration.validate_integration_name!(type)
  ...do stuff...
end
```